analyses/transcriptomic-dimension-reduction/01-transcriptomic-analysis-prep.Rmd

```diff
@@ -0,0 +1,316 @@
+- `analysis/transcriptomic-dimension-reduction/results/kallisto_pca_model.RDS`
+- `analysis/transcriptomic-dimension-reduction/results/kallisto_tsne_model.RDS`
+- `analysis/transcriptomic-dimension-reduction/results/kallisto_umap_model.RDS`
+
+_Note that the tsv files holding the dimension reduction scores represent the scores for the RSEM (denoted by the output tsv files beginning with 15.17) and Kallisto expression data (denoted by the output tsv files beginning with 109.299)._
+
+# Usage
+
+This script is intended to be run via the command line from the top directory of the repository as follows: 
+```
+Rscript -e "rmarkdown::render('analyses/transcriptomic-dimension-reduction/01-transcriptomic-analysis-prep.Rmd', clean = TRUE)"
+```
+
+# Set Up
+
+Set seed for reproducibility.
+```{r}
+set.seed(2019)
+```
+
+Install packages.
+```{r, warning = FALSE, message = FALSE}
+# This is needed for running the t-SNE analysis
+if (!("Rtsne" %in% installed.packages())) {
+  install.packages("Rtsne")
+}
+# This is needed for running the umap analysis
+if (!("umap" %in% installed.packages())) {
+  install.packages("umap")
+}
+```
+
+Assign functions. 
+```{r}
+# magrittr pipe
+`%>%` <- dplyr::`%>%`
+
+perform_dimension_reduction <- function(transposed_expression_matrix, method, seed = 2019) {
+  # Given a data.frame that containes the transposed expression matrix and the
+  # name of a dimension reduction method, perform the dimension reduction 
+  # technique on the transposed matrix
+  # Args:
+  #   transposed_expression_matrix: transposed `RSEM` or `Kallisto` expression
+  #                                 matrix
+  #   method: dimension reduction method to be performed
+  #   seed: seed set to a default of 2019
+  #
+  # Returns:
+  #   dimension_reduction_df: data.frame containing the resulting scores of the
+  #                           dimension reduction technique, with a column that
+  #                           includes sample identifiers
+  
+  # Set seed for reproducibility
+  set.seed(seed)
+  
+  # Save rownames as a vector
+  ID <- rownames(transposed_expression_matrix)
+  # Perform dimension reduction 
+  if (method == "PCA" ){
+  prcomp_results <- prcomp(transposed_expression_matrix)
+  dimension_reduction_df <- data.frame(prcomp_results$x[, 1:2])
+  } else if (method == "t-SNE" ) {
+  tsne_results <- Rtsne::Rtsne(transposed_expression_matrix, perplexity = 15)
+  dimension_reduction_df <- data.frame(tsne_results$Y)
+  } else if (method == "UMAP" ) {
+  umap_results <- umap::umap(transposed_expression_matrix)
+  dimension_reduction_df <- data.frame(umap_results$layout)
+  } else {
+    print ("Error")
+  }
+  
+  # To set the tsv filename 
+  matrix_name <- as.name(transposed_expression_matrix)
+  # Assign the relevant rownames(Kids_First_Biospecimen_ID) to a column
+  dimension_reduction_df$Kids_First_Biospecimen_ID <- ID
+  # Write the resulting data.frame to a tsv file
+  readr::write_tsv(dimension_reduction_df, 
```

Comment:
I would expect the file written by this function would be the entire model as an RDS object. I would also not use the `results_dir` from the global environment but instead pass the filename (which contains the full path information) to the function.